### PR TITLE
Deployment table (deployment data only)

### PR DESCRIPTION
**Pull request by philipnrmn**
This PR depends on functionality introduced in #136. 

It introduces a table which shows deployments data. 

The deployments table final design looks like this:

This PR brings it to the following state:

The eventual intent is to have a separate table row for each deployment and each app which is affected by that deployment. In the interests of minimal changes per PR, only the deployment rows are implemented at present. Status bars and `Rollback` links will be implemented separately.

### CODE REVIEW

**Comment by mesosphere-web** [comment]
Reviewer: @rcorral

File: src/js/pages/services/DeploymentsTab.js [hunk 1]
```diff
@@ -1,11 +1,73 @@
...
 import React from 'react';
+import {Table} from 'reactjs-components';
 
 import AlertPanel from '../../components/AlertPanel';
+import EventTypes from '../../constants/EventTypes';
+import MarathonStore from '../../stores/MarathonStore';
+import ResourceTableUtil from '../../utils/ResourceTableUtil';
+import StringUtil from '../../utils/StringUtil'
+
+const columnClassName = ResourceTableUtil.getClassName;
```
Comment (orlandohohmeier) [inline src/js/pages/services/DeploymentsTab.js]: Please rename the const, as the name kinda suggests that its simply holding class name (`string`) but it's actually a reference to the class name getter. 
BTW: Do we really need to cache this? I don't see any real performance improvements in doing so.

File: src/js/pages/services/DeploymentsTab.js [hunk 2]
```diff
@@ -1,11 +1,73 @@
...
     super(...arguments);
+    this.onDeploymentsChange = this.onDeploymentsChange.bind(this);
+
+    this.state = {
+      deployments: MarathonStore.get('deployments')
+    };
+  }
+
+  componentDidMount() {
+    MarathonStore.addChangeListener(
```
Comment (orlandohohmeier) [inline src/js/pages/services/DeploymentsTab.js]: Not part of this PR, but it would be cool, if we could get the data from the DCOStore instead, to have components only deal with one unified store.

File: src/js/pages/services/DeploymentsTab.js [hunk 3]
```diff
@@ -19,8 +81,34 @@ class DeploymentsTab extends React.Component {
     );
   }
 
+  renderPopulated() {
+    let state = this.state;
+    let deployments = state.deployments.getItems();
```
Comment (orlandohohmeier) [inline src/js/pages/services/DeploymentsTab.js]: Let's use a destructuring to get the deployments from the current state.

**Comment by orlandohohmeier** [comment]
@philipnrmn Looks quite good, besides a few nits. Solid work!

File: src/js/pages/services/DeploymentsTab.js (hunk 1 above)
Comment (philipnrmn) [inline src/js/pages/services/DeploymentsTab.js]: Yeah I wondered about this. It's less a cache and more a set of statics. 

I was using `ServicesTable` as a reference implementation and this is one of the deviations I made. It seemed rather odd to me to recreate the whole object, complete with methods, on every render. 

I agree that from the user's perspective there is zero difference either way.

File: src/js/pages/services/DeploymentsTab.js (hunk 2 above)
Comment (philipnrmn) [inline src/js/pages/services/DeploymentsTab.js]: Yep, working towards that in the long run.

File: src/js/pages/services/DeploymentsTab.js (hunk 3 above)
Comment (philipnrmn) [inline src/js/pages/services/DeploymentsTab.js]: I did this, but I really dislike the resulting code.

**Comment by philipnrmn** [comment]
Addressed the nits and fixed up.

**Comment by philipnrmn** [comment]
No idea why those tests were failing...

*walks away whistling innocently*

**Comment by orlandohohmeier** [comment]
All test passed, let's 🚢 🇮🇹

File: src/js/pages/services/DeploymentsTab.js [hunk 4]
```diff
@@ -1,11 +1,73 @@
...
+];
 
 class DeploymentsTab extends React.Component {
 
   constructor() {
     super(...arguments);
+    this.onDeploymentsChange = this.onDeploymentsChange.bind(this);
+
+    this.state = {
+      deployments: MarathonStore.get('deployments')
```
Comment (pierlo-upitup) [inline src/js/pages/services/DeploymentsTab.js]: I find it quite unsettling that we're assigning the Deployments List by reference to our state – we have no protection from external mutations. That's a generic issue with our Structs approach, not a problem in this PR per se.
